src/VaultToken.sol

```diff
@@ -98,4 +98,13 @@ abstract contract VaultToken is IVaultToken {
 
         emit Transfer(msg.sender, sender, address(0), currency, amount);
     }
+
+    function _burnFrom(address from, Currency currency, uint256 amount) internal virtual {
+        if (msg.sender != from && !isOperator[from][msg.sender]) {
+            uint256 allowed = allowance[from][msg.sender][currency];
+            if (allowed != type(uint256).max) allowance[from][msg.sender][currency] = allowed - amount;
+        }
+
+        _burn(from, currency, amount);
```

Comment:
It seems that directly using the codes in the _burn function, will save some gas?

// _burn(from, currency, amount);
balanceOf[from][currency] -= amount;
emit Transfer(msg.sender, from, address(0), currency, amount);

Reply:
It seems not a big deal in this case probabaly a extra jump ?  I feel like `_burn` gives better readability

![CleanShot 2024-04-09 at 16 51 32@2x](https://github.com/pancakeswap/pancake-v4-core/assets/137024020/f4f2a696-d40b-4c9a-9c0b-cc372d9d2384)